components/optional/index.ts

```diff
@@ -1,7 +1,15 @@
 import PropTypes from 'prop-types';
 import { useBlockEditContext } from '@wordpress/block-editor';
 
-export const Optional = ({ value, children }) => {
+interface OptionalProps {
+	value?: string | number | boolean;
+	children: React.ReactNode;
+}
+
+export const Optional: React.FC<OptionalProps> = ({
+	value,
+	children,
+}): boolean | React.ReactNode => {
```

Comment:
I'm not sure we need / want to explicitly type the return type of every function. If something can be inferred I would rather just leave it up to the compiler